### DISCUSSION TITLE
how to remap key ?

### DISCUSSION THREAD

**Discussion by WingDust** [Genivia/ugrep#367] [Q&A] [answered]
I want to remap `ctrl-q` to `ctrl-o` in ugrep interactive mode, how to deal with ?

Also how ugrep  work like `rg --column . | fzf --preview 'echo {1}  '`  , right side  `ctrl-n` and `ctrl-p` select file and input char fuzzy match file  , `Enter ` stdout ouput like `temp4.log:1:1:2` without manual select search result
`

**Comment by genivia-inc** [comment, marked as answer]
Remapping TUI keys is not possible. Also ESC or CTRL-C work to quit (turn confirmation off in a .ugrep file with `no-confirm` when desired). If there are use cases that require remapping keys, then this can be requested as a new feature.

For the second question, ugrep is not meant to be a replacement for fzf, although fuzzy search is supported by ugrep and the TUI supports file (pre)views. In the TUI CTRL-Y views a selected file with `more` or `less` as a pager, in which you can specify search patterns. Other viewers can be assigned by specifying `view=PAGER` in the .ugrep file.

If you want to first recursively select files matching a PATTERN and then search the selected files with various (fuzzy) patterns, then use `ug -l PATTERN | xargs ug -Q -Z`.

It is unclear what you mean by "char fuzzy match file , `Enter`  stdout ouput like `temp4.log:1:1:2` without manual select search result".

A way to get fuzzy-matching files sorted by best match to view, you could do
```console
$ ug -lZQ --sort=best
```
and then type in patterns to preview the file(s) shown in the preview screen (CTRL-T) or view them with CTRL-Y. Press CTRL-Z for help and to change search options. Press ALT-g to enter globs to select files and dirs.
